Control the use of the cl4py readtable.

rpgoldman opened the pull request:
The cl4py readtable is wrapped around *everything* done by Lisp running inside cl4py, instead of just being wrapped around reading from the stream of inputs from python.

In particular, this means that any lisp code compiled by cl4py is read with the cl4py readtable, which can cause breakage if the standard readtable is expected.

This tweak makes the cl4py readtable apply only around the `read` rather than the `read` and the `eval`, as before.

rpgoldman committed:
Control the use of the cl4py readtable.

It was leaking state into the compiler, and breaking lots of code.

marcoheisig commented:
This is a great suggestion.  I will try to merge, test and upload the fix this week.

rpgoldman commented:
Probably a test would be a good idea -- is `pytest` ok? I have never used `pyunit`, and am not really interested in learning a new framework.

marcoheisig commented:
Sure, any tests would be much appreciated.  With some embarrassment I have to admit that, so far, my tests consisted of me running the examples in the README by hand before uploading a new version to PyPi.

rpgoldman commented:
Unfortunately, I really don't know how Travis works or how to set up tests that would involve SBCL.  But I could see about pytest, and maybe you or someone else could make it work with Travis.

rpgoldman committed:
Add equality method for Cons.

This is needed to be able to write the tests.

rpgoldman committed:
Tests from README.rst

New pytests. Added before adding tests for this MR.

rpgoldman committed:
Test for MR #9 issue.

Verify that the previous version of cl4py would experience errors because the python readtable was used to read *all* input to the lisp process (instead of only input from the Python process).

rpgoldman commented:
OK, now there are tests!

marcoheisig commented:
Merged.  Thank you for all those contributions!

rpgoldman commented:
> Merged. Thank you for all those contributions!

Glad I could help! Thank _you_ for the library!
